pkgs/build-support/setup-hooks/patch-shebangs.sh

```diff
@@ -151,14 +126,31 @@ patchShebangs() {
 
                 # Preserve times, see: https://github.com/NixOS/nixpkgs/pull/33281
                 timestamp=$(stat --printf "%y" "$f")
-                sed -e "1 s|.*|#\!$escapedInterpreterLine|" "$f" | _sponge "$f"
+
+                # Manually create temporary file instead of using sed -i
+                # (sed -i on $out/x creates tmpfile /nix/store/x which fails on macos + sandbox)
+                tmpFile=$(mktemp -t patchShebangs.XXXXXXXXXX)
```

Comment:
We never clean this up. Don't know if this is an issue. Potentially we could also create one directory that is used and than always write to the same file in this directory?

Reply:
But potentially this is also over engineered.